src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -1028,13 +2063,26 @@ def gen_code(self, parent):
 
         calls = self.reductions()
         zero_reduction_variables(calls, parent)
-        private_str = ",".join(self._get_private_list())
+        private_clause = self._get_private_list()
+        if len(self._children) >= 3 and private_clause != self._children[2]:
```

Comment:
Is this just ensuring that the PSyIR is self-consistent at this point?

Reply:
Yes, and make sure the private clause is in the right place/doesn't remove another clause etc.